### PR TITLE
feat(hide-on-scroll): add component #339

### PR DESCRIPTION
**Pull request by mart-sessman**

### CODE REVIEW

**Comment by kasparsuvi1** [comment]
HideOnScroll should be separate PR and also delete commmunity version of it?

**Comment by mart-sessman** [comment]
Yes, Floating Button should not be in this PR. Fixing this.

Also renaming this HideOnScroll component to ScrollVisibility, because it does more than just hiding, like an example: showing content when scrolling past scroll distance. But this will be done in separate task.

Since this component's name won't be "HideOnScroll", the community version will stay, but will be deprecated and new component linked to it.

**Comment by mart-sessman** [comment]
You are right, keeping 2 components with the same functionality is not reasonable.
Removed old component and old hook.
Also there is one BREAKING CHANGE.

Old component behaved this way:
When scrolled past "scrollDistance" prop and then scrolled back up 10px, the content would show again.
This behaviour is still present, but disabled by default.
Now for the same behaviour, "toggleVisibility" prop must be set to true.

File: libs/react-components/src/tedi/components/hide-on-scroll/hide-on-scroll.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+const CONTAINER_HEIGHT = 300;
+
+/**
+ * <a href="https://tedi.tehik.ee/1ee8444b7/p/00ab5a-scrollvisibility-dev" target="_BLANK">Zeroheight ↗</a>
+ **/
+
+const meta: Meta<typeof HideOnScroll> = {
+  component: HideOnScroll,
+  title: 'Tedi-ready/Components/HideOnScroll/HideOnScroll',
```
Comment (airikej) [inline libs/react-components/src/tedi/components/hide-on-scroll/hide-on-scroll.stories.tsx:17]: This component could be placed inside Helpers category, at least for now

**Review by airikej — APPROVED** [review]

**Pull request merged**